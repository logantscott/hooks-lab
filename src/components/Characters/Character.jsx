import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ _id, name, affiliation }) => (
  <p>{name} - {affiliation}</p>
);

Character.propTypes = {
  _id: PropTypes.string.isRequired,
  photoUrl: PropTypes.string,
  name: PropTypes.string.isRequired,
  affiliation: PropTypes.string
};

export default Character;
