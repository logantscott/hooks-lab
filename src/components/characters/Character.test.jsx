import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('Character component', () => {
  it('renders Character', () => {
    const wrapper = shallow(<Character _id="5cf5679a915ecad153ab68ca" name="Aang" affiliation="Air Nomads Team Avatar" />);
    expect(wrapper).toMatchSnapshot();
  });
});
