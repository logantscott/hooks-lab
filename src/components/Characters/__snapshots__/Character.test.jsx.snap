// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<p>
  Aang
   - 
  Air Nomads Team Avatar
</p>
`;
